storage/innobase/xtrabackup/src/xbcloud/xbcloud.cc

```diff
@@ -705,9 +705,20 @@ bool xbcloud_put(Object_store *store, const std::string &container,
     return false;
   }
 
-  // check if xtrabackup_info, almost the last file exists in backup
-  if (std::count(object_list.begin(), object_list.end(),
-                 backup_name + "/xtrabackup_info.00000000000000000000") == 0) {
+  /* check if the last_file (xtrabackup_tablespaces.gz.00000000000 or
+  xtrabackup_tablespaces.00000000000000000000) is uploaded to cloud storage to
+  determine successful xbcloud "put" operation. */
+  std::string last_file_prefix = backup_name + "/xtrabackup_tablespaces";
+  bool file_found = false;
+
+  if (!object_list.empty()) {
+    auto cur_file = object_list.back();
+    auto last_file_size = last_file_prefix.size();
+    if (cur_file.size() >= last_file_size &&
+        cur_file.substr(0, last_file_size).compare(last_file_prefix) == 0)
+      file_found = true;
```

Comment:
I think we should test the upload on a per-file basis. I don't think what it's been done here can ensure you have a good backup. Here we only ensure that xtrabackup_tablespaces/xtrabackup_info has been uploaded, but what about all the other files? 
Perhaps we should at least compare the number of files on remote dst against the number of files we expect based on a runtime calculation from copy threads.

Reply:
xbcloud is different api. so the fix I am thinking is to parse the total number of files written by xtrabackup to xbcloud API and then verify that. for that, we also have to consider additional files written by backup based on the option provided which need some additional testing as well. 

this bug just fixes the regression where we are checking xtrabackup_info instead of xtrabackup_info.gz
IMHO we can create a separate bug and implement these changes with the best approach.

Reply:
Agree on making it right to consider compression and raise an enhancement issue to make it more reliable.